src/device_trezor/device_trezor_base.hpp

```diff
@@ -227,9 +227,9 @@ namespace trezor {
         }
 
         if (network_type){
-          msg->set_network_type(static_cast<uint32_t>(network_type.get()));
+          msg->set_network_type(static_cast<messages::monero::MoneroNetworkType>(network_type.get()));
         } else {
-          msg->set_network_type(static_cast<uint32_t>(this->network_type));
+          msg->set_network_type(static_cast<messages::monero::MoneroNetworkType>(this->network_type));
```

Comment:
Having `network_type` as a function parameter and as a class member is confusing, which is why you had to write `this->network_type` here. It's better to rename the class member to `m_network_type` to avoid further confusion.